pytensor/tensor/nlinalg.py

```diff
@@ -595,6 +596,89 @@ def infer_shape(self, fgraph, node, shapes):
         else:
             return [s_shape]
 
+    def L_op(
+        self,
+        inputs: Sequence[Variable],
+        outputs: Sequence[Variable],
+        output_grads: Sequence[Variable],
+    ) -> list[Variable]:
+        """
+        Reverse-mode gradient of the SVD function. Adapted from the autograd implementation here:
+        https://github.com/HIPS/autograd/blob/01eacff7a4f12e6f7aebde7c4cb4c1c2633f217d/autograd/numpy/linalg.py#L194
+
+        And the mxnet implementation described in ..[1]
+
+        References
+        ----------
+        .. [1] Seeger, Matthias, et al. "Auto-differentiating linear algebra." arXiv preprint arXiv:1710.08717 (2017).
+        """
+        (A,) = (cast(ptb.TensorVariable, x) for x in inputs)
+
+        if not self.compute_uv:
+            # We need all the components of the SVD to compute the gradient of A even if we only use the singular values
+            # in the cost function.
+            U, s, VT = svd(A, full_matrices=False, compute_uv=True)
+
+            (ds,) = (cast(ptb.TensorVariable, x) for x in output_grads)
+            A_bar = (U.conj() * ds[..., None, :]) @ VT
+
+            return [A_bar]
+
+        elif self.full_matrices:
+            raise NotImplementedError(
+                "Gradient of svd not implemented for full_matrices=True"
+            )
+
+        else:
+            U, s, VT = (cast(ptb.TensorVariable, x) for x in outputs)
+            (dU, ds, dVT) = (cast(ptb.TensorVariable, x) for x in output_grads)
```

Comment:
This is interesting. What happens when the gradient wrt to one of these outputs is zeros or undefined (e.g., because it goes through some non-differentiable Op later)? Does this behave correctly?

Reply:
I assume we get a `DisconnectedInput` error in those cases. If a component isn't used in the cost computation at all, I think we want the gradient contribution here to be zero. If it goes into a black box, though, I think the error is correct.

Reply:
Cool, are we handling those cases already correctly?